client/core/src/com/cows/game/roundSimulation/simulationModels/TowerSimulationModel.kt

```diff
@@ -0,0 +1,58 @@
+package com.cows.game.roundSimulation.simulationModels
+
+
+import com.cows.game.map.Coordinate
+
+class TowerSimulationModel(val id: Int, val position: Coordinate, range: Int, unitPath: List<IntArray>, private val timeBetweenAttacks: Int, private val damage: Int) {
+    var target: UnitSimulationModel? = null
+    var cooldown: Int = 0
+    var pathIndicesInRange = findPathIndicesInRange(range, unitPath)
+
+
+    fun decrementCooldown() {
+        if (cooldown > 0) {
+            cooldown--
+        }
+    }
+
+    private fun setCooldown() {
+        cooldown = timeBetweenAttacks
+    }
+
+    fun attack() {
+        target!!.damage(damage)
```

Comment:
Is target guaranteed to be not null?

Reply:
You can use the _let_ syntax like in the _targetInRange_ method.

Reply:
It is in theory guaranteed at an earlier stage, but I agree that it should be a let statement